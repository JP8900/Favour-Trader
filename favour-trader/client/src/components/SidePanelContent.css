.SidePanelContent {
    margin-top: 45px;
    overflow: hidden;
    margin-left: 10px;
}

.SidePanelContent-btn {
    background-color: white;
    color: black;
    border-color: black;
    border: none;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    border-radius: 3px;
    margin-top: 5px; 
    box-shadow: none !important;
}

ul {
    padding-left: 20px;
}