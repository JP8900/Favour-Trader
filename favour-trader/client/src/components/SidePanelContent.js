import React, { Component } from 'react';
import axios from 'axios'
import './SidePanelContent.css';

class SidePanelContent extends Component {
    constructor() {
        super();
        this.state = {
            allSkills: [],
            hasSkills: [],
            wantSkills: [],
        };
    }

    componentDidMount() {
        const { authService } = this.props;
        if (authService.loggedIn()) {
            const userProfile = authService.getProfile();
            const config = {
               headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
                'Authorization': authService.getToken(),
                }
            };

            axios.get('/api/skills/all')
            .then(res => res.data)
            .then(data => this.setState({allSkills: data }));

            if (!!userProfile && userProfile.email) {
                const userEmail = userProfile.email;
                axios.post('/api/users/has', {
                    email: userEmail,
                }, config)
                .then(res => res.data.user)
                .then(user => this.setState({ hasSkills: user.has }));

                axios.post('/api/users/wants', {
                    email: userEmail,
                }, config)
                .then(res => res.data.user)
                .then(user => this.setState({ wantSkills: user.wants }));
            }
        }
    }

    renderSkills(skillSet) {
        const skills = this.state[skillSet];
        return (
            <div>
                {
                    (skills !== []) ? (
                        <ul>
                            {
                                skills.map(function (skill) {
                                    return (<li key={skill._id}> {skill.skill} </li>)
                                }, this)
                            }
                        </ul>
                    ) : (
                            ''
                    )
                }
            </div>
        );
    }

    render() {
        return (
            <div className={'SidePanelContent'}>
                Skills I Have:
                <div>
                {
                    this.renderSkills('hasSkills')
                }
                <button className={'SidePanelContent-btn'}>Add a new skill</button>
                </div>
                <hr/>
                Skills I'm Looking for:
                <div>
                {
                    this.renderSkills('wantSkills')
                }
                <button className={'SidePanelContent-btn'}>Add a new skill</button>
                </div>
                <hr/>
                All Skills:
                <ul>
                {
                    this.renderSkills('allSkills')
                }
                </ul>
                <hr/>
            </div>
        );
    }
}

export default SidePanelContent;