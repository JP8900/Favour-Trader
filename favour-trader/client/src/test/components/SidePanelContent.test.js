import React from 'react';
import renderer from 'react-test-renderer';
import SidePanelContent from '../../components/SidePanelContent.js';
import AuthServiceMock from './AuthServiceMock.js';

it('renders correctly', () => {
    var authServiceMock = new AuthServiceMock();
    const tree = renderer
        .create(<SidePanelContent authService={authServiceMock} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});