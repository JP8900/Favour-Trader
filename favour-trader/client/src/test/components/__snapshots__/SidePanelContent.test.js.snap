// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="SidePanelContent"
>
  Skills I Have:
  <div>
    <div>
      <ul />
    </div>
    <button
      className="SidePanelContent-btn"
    >
      Add a new skill
    </button>
  </div>
  <hr />
  Skills I'm Looking for:
  <div>
    <div>
      <ul />
    </div>
    <button
      className="SidePanelContent-btn"
    >
      Add a new skill
    </button>
  </div>
  <hr />
  All Skills:
  <ul>
    <div>
      <ul />
    </div>
  </ul>
  <hr />
</div>
`;
